src/holon/rule_engine/repositories/repository_base.py

```diff
@@ -234,15 +233,24 @@ def remove(self, object_id: int):
         list_index = self.get_list_index_for_object_id(object_id)
         self.objects.pop(list_index)
 
-    def assert_correct_object_type(self, object: PolymorphicModel):
+    def distinct(self, attributes: list[str]) -> list[dict]:
+        # first create set so only unique combinations remain
+        set = {
+            tuple(map(lambda attribute: getattr(s, attribute), attributes)) for s in self.objects
+        }
+
+        # then convert to list of dictionaries so caller code is more readable
+        return [
+            {attribute: value_tuple[i] for i, attribute in enumerate(attributes)}
+            for value_tuple in set
+        ]
```

Comment:
Docstring mist nog.
Verder dacht ik dat deze methode de distincte objecten gegroupeerd op de gegeven attributes zou teruggeven, en niet alleen de unieke combinaties van attribute values aanwezig in de objects lijst. 
Als dit wel zo hoort zou ik voorstellen om de functienaam te veranderen naar iets van `get_distinct_attribute_values` o.i.d., om verwarring te voorkomen met bijv. de Django `.distinct` werking, die wel objecten teruggeeft.

Reply:
Hernoemd. Ik vind de django functionaliteit nogal vreemd, dat je at random een object krijgt.